cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -427,8 +438,22 @@ def __pow__(self, power: int) -> 'CircuitOperation':
     def _with_key_path_(self, path: Tuple[str, ...]):
         return dataclasses.replace(self, parent_path=path)
 
-    def _with_key_path_prefix_(self, prefix: Tuple[str, ...]):
-        return dataclasses.replace(self, parent_path=prefix + self.parent_path)
+    def _with_key_path_prefix_(
+        self,
+        path: Tuple[str, ...],
+        local_keys: FrozenSet[value.MeasurementKey],
+        extern_keys: FrozenSet[value.MeasurementKey],
+    ):
+        new_keys = set(extern_keys)
```

Comment:
I think there may be a bug here if there are no repetitions. In that case the local keys that get passed in should remain local I think. Because even though you're in a subcircuit, the scope itself doesn't change.